storage/redis/redisserver/server.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+// Package redisserver is package for starting a redis test server
+package redisserver
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/alicebob/miniredis"
+	"github.com/go-redis/redis"
+)
+
+const (
+	fallbackAddr = "localhost:6379"
+	fallbackPort = 6379
+)
+
+func freeport() (addr string, port int) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return fallbackAddr, fallbackPort
+	}
+
+	netaddr := listener.Addr().(*net.TCPAddr)
+	addr = netaddr.String()
+	port = netaddr.Port
+	_ = listener.Close()
+	time.Sleep(time.Second)
+	return addr, port
+}
+
+// Start starts a redis-server when available, otherwise falls back to miniredis
+func Start() (addr string, cleanup func(), err error) {
+	addr, cleanup, err = Process()
+	if err != nil {
+		log.Println("failed to start redis-server: ", err)
+		return Mini()
+	}
+	return addr, cleanup, err
+}
+
+// Process starts a redis-server test process
+func Process() (addr string, cleanup func(), err error) {
+	tmpdir, err := ioutil.TempDir("", "storj-redis")
+	if err != nil {
+		return "", nil, err
+	}
+
+	// find a suitable port for listening
+	var port int
+	addr, port = freeport()
```

Comment:
hmm, bummer we can't just tell redis to choose one and then get the port it chose somehow

Reply:
Unfortunately no, redis seems to use ":0" to disable the TCP port.